C7GameData/SaveFormat.cs

```diff
@@ -10,27 +10,59 @@ namespace C7GameData
 */
 {
     using System.IO;
+    using System.IO.Compression;
     using System.Text.Json;
+
+    public enum SaveFileFormat
+    {
+        Plain,
+        Zip,
+    }
+
     public class C7SaveFormat
     {
         public string Version = "v0.0early-prototype";
+
         // Rules is intended to be the analog to a BIC/X/Q
         public C7RulesFormat Rules;
+
         // This naming is probably bad form, but it makes sense to me to name it as such here
         public GameData GameData;
+
         public C7SaveFormat()
         {
 	        GameData = new GameData();
         }
+
         public C7SaveFormat(GameData gameData, C7RulesFormat rules = null)
         {
-            this.GameData = gameData;
+            GameData = gameData;
             Rules = rules;
         }
-        public static C7SaveFormat Load(string path)
+
+        public static C7SaveFormat Load(string path, SaveFileFormat format = SaveFileFormat.Plain)
         {
-            string json = File.ReadAllText(path);
-            C7SaveFormat save = JsonSerializer.Deserialize<C7SaveFormat>(json, JsonOptions);
+            System.Console.WriteLine("loading...");
+            C7SaveFormat save = null;
+            if (format == SaveFileFormat.Plain)
+            {
+                string json = File.ReadAllText(path);
+                save = JsonSerializer.Deserialize<C7SaveFormat>(json, JsonOptions);
+            }
+            else
+            {
+                using (var fileStream = new FileStream(path, FileMode.Open))
+                using (var archive = new ZipArchive(fileStream, ZipArchiveMode.Read))
```

Comment:
Does it cause a problem to inline the FileStream?  I don't know if that's idiomatic in C# like it is in Java, but given the amount of nested using statements it seems like it would read better.

Reply:
I'm not sure if `FileStream.Dipose` gets called if it's inlined, but I'll double check

Reply:
looks like it might be idiomatic, seeing as `ZipArchive` supports this: [docs](https://docs.microsoft.com/en-us/dotnet/api/system.io.compression.ziparchive.dispose?view=net-6.0#:~:text=Unless%20you%20construct%20the%20object,available%20for%20subsequent%20write%20operations.)